modules/features/app-widgets/src/main/java/com/duchastel/simon/simplelauncher/features/appwidgets/ui/compose/AppWidgetHostCompose.kt

```diff
@@ -0,0 +1,109 @@
+package com.duchastel.simon.simplelauncher.features.appwidgets.ui.compose
+
+import android.appwidget.AppWidgetHostView
+import android.appwidget.AppWidgetManager
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.size
+import androidx.compose.material3.CircularProgressIndicator
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.DisposableEffect
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.viewinterop.AndroidView
+import com.duchastel.simon.simplelauncher.features.appwidgets.data.WidgetData
+import com.duchastel.simon.simplelauncher.features.appwidgets.host.LauncherAppWidgetHost
+
+@Composable
+fun AppWidgetHostCompose(
+    widgetData: WidgetData,
+    appWidgetHost: LauncherAppWidgetHost,
+    modifier: Modifier = Modifier,
+    onError: ((Throwable) -> Unit)? = null
+) {
+    val context = LocalContext.current
+    var hostView by remember(widgetData.widgetId) { mutableStateOf<AppWidgetHostView?>(null) }
+    var isLoading by remember(widgetData.widgetId) { mutableStateOf(true) }
+    var error by remember(widgetData.widgetId) { mutableStateOf<Throwable?>(null) }
+
+    LaunchedEffect(widgetData.widgetId) {
+        try {
+            val appWidgetManager = AppWidgetManager.getInstance(context)
+            val providerInfo = appWidgetManager.getAppWidgetInfo(widgetData.widgetId)
+            
+            if (providerInfo != null) {
+                val view = appWidgetHost.createView(context, widgetData.widgetId, providerInfo)
+                hostView = view
+                isLoading = false
+            } else {
+                throw IllegalStateException("Widget provider info not found for ID: ${widgetData.widgetId}")
```

Comment:
Do no throw exceptions. This should be typed errors properly bubbled up and handled (moving this to the data layer and/or circuit screen presenter will fix this).